src/lib/support/BUILD.gn

```diff
@@ -108,9 +109,15 @@ static_library("support") {
     "${chip_root}/src/inet:inet_config_header",
   ]
 
+  # Android has no 'platform' that provides logging
+  if (current_os == "android") {
+    public_deps += [ "${chip_root}/src/platform/logging:android" ]
+  }
+
   public_configs = [
     "${chip_root}/src:includes",
     "${chip_root}/src/lib:includes",
+    "${chip_root}/src/platform/logging:includes",
```

Comment:
This is wrong, logging should be pulled in via deps and we should not have to add include paths directly - they should be added to the target that contains the headers as public_configs and propagated via deps.

Reply:
I was not sure how to break the circular dependency:

- logging output requires 'category' (and that is not platform-specific, so it makes sense to have it central - currently in support)
- support/logging requires platform specific logging

I thought about making the category header as a separate source set, however the logging implementation is platform-dependent (and platform code does logging).

So I have:
   - platforms needs to log
   - logging needs platform-dependent code

Reply:
> I was not sure how to break the circular dependency:
> 
> * logging output requires 'category' (and that is not platform-specific, so it makes sense to have it central - currently in support)
> * support/logging requires platform specific logging
> 
> I thought about making the category header as a separate source set, however the logging implementation is platform-dependent (and platform code does logging).
> 
> So I have:
> 
> * platforms needs to log
> * logging needs platform-dependent code

Generally I would recommend to break cycles in a way that allows unit tests to link without additional dependencies.

This way would not allow libSupport's unit tests to link without an extra src/platform dependency, except on Android. We do have such a dependency from those tests, so it works, but it is an undeclared dependency of src/lib/support and undeclared dependencies should be avoided.

You can have a

```
source_set("logging_constants") {
  sources = ["logging/constants.h"]
}
```

It doesn't need to be in support to be "common".

Reply:
Updated to use source sets. I still have files in support using platform and platform using support, but it it is not circular anymore, just cross-referenced.